### PR TITLE
[READY FOR REVIEW] MONGOID-5570 Code Docs: 100% class/method documentation coverage 💯🎉 

### PR DESCRIPTION
**Pull request by johnnyshields**
This PR adds 100% coverage of class and method documentation.

This PR contains only doc comments; code was modified.

It also enables the following Rubocop cops, which will ensure coverage stays at 100%:
- [Style/Documentation](https://docs.rubocop.org/rubocop/cops_style.html#styledocumentation)
- [Style/DocumentationMethod](https://docs.rubocop.org/rubocop/cops_style.html#styledocumentationmethod)

### CODE REVIEW

File: gemfiles/standard.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def standard_dependencies
 
   group :development, :test do
     gem 'rspec-core', '~> 3.10'
-    gem 'rubocop', '~> 1.45.1'
+    gem 'rubocop', git: 'https://github.com/rubocop/rubocop.git'
```
Comment (johnnyshields) [inline gemfiles/standard.rb:23]: Can be '~> 1.45.2' when released.

**Comment by alexbevi** [comment]
@johnnyshields this is a change we would prefer to introduce incrementally. MONGOID-5304 was merged and introduces static analysis, however we'd like to take some time to evaluate our options before enabling additional cops.

I'm going to close this PR and MONGOID-5570 for the time being and we'll add additional tickets to the backlog as we introduce additional cops.

**Comment by johnnyshields** [comment]
@alexbevi would you accept this PR if I remove the Rubocop enablement? Documenting all these methods took a considerable amount of time and research and it would be a shame for it to go to waste.

**Comment by johnnyshields** [comment]
@alexbevi I've raised https://github.com/mongodb/mongoid/pull/5559 with the Rubocop changes removed.

**Comment by alexbevi** [comment]
> @alexbevi I've raised #5559 with the Rubocop changes removed.

I'll discuss with the team, however we would need to validate each entry for accuracy as well so the time to review and merge such a large PR would not be insignificant. We may consider "peeling off" chunks of the PR incrementally as those code files are updated instead.

**Comment by johnnyshields** [comment]
Peeling off chunks is fine by me!

**Pull request closed without merging**